Library/Homebrew/download_strategy.rb

```diff
@@ -71,25 +71,25 @@ def quiet?
   # Unlike {Resource#stage}, this does not take a block.
   #
   # @api public
-  def stage
+  def stage(&block)
     UnpackStrategy.detect(cached_location,
                           prioritise_extension: true,
                           ref_type: @ref_type, ref: @ref)
                   .extract_nestedly(basename:             basename,
                                     prioritise_extension: true,
                                     verbose:              verbose? && !quiet?)
-    chdir
+    chdir(&block)
   end
 
-  def chdir
+  def chdir(&block)
     entries = Dir["*"]
     raise "Empty archive" if entries.length.zero?
-    return if entries.length != 1
+    return yield if entries.length != 1
```

Comment:
This API is public, so we need to allow it work without a block (for now at least).

```suggestion
    if entries.length != 1
      yield if block_given?
      return
    end
```

Reply:
Yep, thanks. I've been struggling locally getting everything to work properly. Just running `brew tests` (hopefully for the last time) then I'll push the changes.